### ISSUE TITLE
Contents of JavaScript files being "shaken and stirred" upon minifying

### ISSUE THREAD

**Issue by Xurk** [JimBobSquarePants/Cruncher#12]
There is an issue that we've noticed on a few of our websites recently, specifically: when we use the `RenderJavaScript` method in a template and apply the `minifyJS` setting in the Cruncher configuration, this results in the generated and minified `js.axd` file mixing up the order of JavaScript files' contents.

An illustration of this issue (only the relevant lines have been included):

**1. The Cruncher configuration:**

``` xml
<cruncher>
  <processing>
    <compression minifyCSS="true" minifyJS="true" compressResponse="true" />
  </processing>
</cruncher>
```

**2. Calling the `RenderJavascript` method in the template:**

``` html
@CruncherBundler.RenderJavaScript(
  "jquery",
  "jquery.validate.js",
  "functions-sitename.js"
)
```

**3. The `<script>` tag on the resulting page**

``` html
 <script type="text/javascript" src="/js.axd?path=jquery|jquery.validate.js|functions-sitename.js"></script>
```

**4. The resulting js.axd file***

``` javascript
/* Comment block for jQuery JavaScript Library */
function doSomething() { $('.contactForm').validate({rules:{Email:{email:true,required:true}}}); }
(function(global,factory){ /* jQuery core code */ }
// ... etc ...
```

The problem which arises here in this simplified example is that one of the functions from `functions-sitename.js` has been moved to the top of the minified `js.axd` file. This function utilises jQuery and attempts to invoke a method in jQuery Validate, but both of these third-party files aren't initialised yet at that point in the file. This throws errors on the page akin to `jquery is undefined` or `$.validate() is not a function`.

If we alter the `minifyJS` attribute in the Cruncher settings to `false` and let Cruncher generate a new file, the generated file maintains the correct order perfectly. Obviously, this is less than ideal as the unminified JavaScript file could be up to three times larger than its counterpart which is a big deal if you're talking about 200 vs 600 KB.

Is this a known issue or is there something we may be missing what could be causing this?

**Comment by Xurk** [comment]
A very belated response, but good to hear that this has been fixed! I am going to try updating a website to the the newest version now and run some tests to see if the bug has indeed disappeared in our implementation.

**Comment by JimBobSquarePants** [comment]
@Xurk Better late than never! :smile:

**Comment by Xurk** [comment]
@JimBobSquarePants Unfortunately, we ran into a hiccup as the LibSass reference is getting in the way when trying to run the website (solution builds fine though). I think it's related to this issue: https://github.com/JimBobSquarePants/Cruncher/issues/28

Seeing as we don't need the Sass component I'm going to look into trying to remove it from the solution but my first attempt wasn't successful as I got an error saying that Cruncher depends on it.

I'll send an update after I've figured it out / next time I work on it.
